pkgs/os-specific/linux/kernel/generic.nix

```diff
@@ -45,8 +53,10 @@ let
   } // features) kernelPatches;
 
   config = import ./common-config.nix {
-    inherit stdenv version extraConfig;
-    kernelPlatform = hostPlatform;
+    inherit stdenv version ;
+    # append extraConfig for backwards compatibility but also means the user can't override the kernelExtraConfig part
+    extraConfig = extraConfig + lib.optionalString (hostPlatform ? kernelExtraConfig ) hostPlatform.kernelExtraConfig;
```

Comment:
@teto great change, I didn't even notice this before :). `kernelExtraConfig` isn't used anywhere, so let's definitely get rid of that!

Reply:
Do you mean moving kernelExtraConfig from platforms.* to the buildLinux calls ?

Reply:
Oh wait, my grep was bad, nevermind.